tests/unit/pubsub_v1/publisher/test_publisher_client.py

```diff
@@ -63,6 +63,25 @@ def test_init_w_api_endpoint():
     ) == "testendpoint.google.com"
 
 
+def test_init_w_unicode_api_endpoint():
+    client_options = {"api_endpoint": u"testendpoint.google.com"}
```

Comment:
We should also check the non-unicode string value (the one that can be problematic in Python 2), i.e. `{"api_endpoint": "testendpoint.google.com"}`.

Reply:
I do that in the test above, no?

Reply:
Ah, of course, late hour. It's the unicode version that would fail the `type(...) is str` check, nevermind.